packages/consent/consent-tools/src/domain/create-wrapper.ts

```diff
@@ -189,4 +191,26 @@ const omitDisabledRemotePlugins = (
     // Enable if all of its consent categories are consented to
     const hasUserConsent = categories.every((c) => consentedCategories[c])
     return hasUserConsent
-  })
+  }
+
+  const results = Object.keys(
+    integrations
+  ).reduce<CDNSettingsWithFilteredIntegrations>(
+    (acc, creationName) => {
+      if (!isPluginEnabled(creationName)) {
+        // remote disabled action destinations
+        acc.remotePlugins = acc.remotePlugins.filter(
+          (el) => el.creationName !== creationName
+        )
+        // remove disabled classic destinations and locally-installed action destinations
+        delete acc.integrations[creationName]
+      }
+      return acc
+    },
+    {
+      remotePlugins,
+      integrations: { ...integrations }, // make shallow copy to avoid mutating original
```

Comment:
nit: to not have to do a shallow copy we could construct the `integrations` part on the reducer since we're just putting everything except the disabled ones.

I'm not sure if every `remotePlugin` will have an `integration` object, if so it could also be constructed as it iterates, but I'm assuming from the way this logic is written that it's not the case.

Reply:
Correct, I am not certain every remotePlugin gets a corresponding integration object.

I went with this because it felt a bit easier for others read, since everything can be nested in the `isPluginEnabled` === false block, whereas with constructing integrations -- that would only happened if `isPluginEnabled` === true -- which felt a little bit of a brain teaser.